Improve support for Fedora AMIs

ctubbsii opened the pull request:
Based on the recent mailing list discussion, I decided to test Muchos with Fedora. It mostly works, but needs some changes. This PR also proposes changing the default AMI for EC2 to a recent Fedora 35 AMI, but that is not critical, and can be reverted before this PR is merged.

arvindshmicrosoft commented on conf/muchos.props.example:
```diff
@@ -30,7 +30,7 @@ install_dir = %(user_home)s/install
 proxy_hostname = leader1
 # If set, a SOCKS proxy will be created on the specified port when connecting to proxy using 'muchos ssh <cluster>'
-#proxy_socks_port = 38585
+proxy_socks_port = 38585
```
Is this change required due to the default OS changing?

ctubbsii replied:
No. I just uncommented it while testing and left it that way because it's super useful. I didn't see a reason why it was commented out by default. I can put it back, but I'm not sure there's a good reason to leave it off by default.

arvindshmicrosoft commented on conf/muchos.props.example:
```diff
@@ -18,7 +18,7 @@
 # Cluster user name (install command will SSH to cluster using this user)
 # Leave default below if launching cluster in AWS
-cluster_user = centos
+cluster_user = fedora
```
To make this neutral and OS-agnostic, can we just use `muchos` instead?

ctubbsii replied:
The default user name depends on the AMI, and is not generally something we choose. The default user for Fedora AMIs is documented to be "fedora", and the default user for CentOS AMIs is documented to be "centos". This is the user whose home directory is injected with the SSH public key specified at launch time. For cloud-init enabled images, this can be overridden by passing cloud-config userdata at launch time, but we're currently not doing that for Muchos, so the configured user has to match the documented default user for the AMI we've selected.

arvindshmicrosoft replied:
Got it. This is also probably a difference between the Muchos `launch` mechanisms for EC2 and Azure. For Azure since we use Ansible itself to launch the VMs, we are able to set the default [admin user name](https://docs.ansible.com/ansible/latest/collections/community/azure/azure_rm_virtualmachinescaleset_module.html#parameter-admin_username) to the `cluster_user` at launch time itself. From what I can see in the Muchos ec2 launch code, we don't pass the `cluster_user` to the EC2 instance launch calls (via. boto3), which is likely why you had to change the value of `cluster_user` instead to match the AMI default.

These differences are all good learnings towards eventually normalizing architecture and code as much as possible between the different cluster types, specifically for the `launch` phase which is pretty much where the cluster-type specific code is. Anyways, that's for another day 😃

ctubbsii replied:
The launch mechanism for EC2 could be modified to set the default user's name in cloud-config. Actually, a lot could be done with cloud-config, rather than relying on ansible for so much... like RPM package installations for first-boot / initial setup, as well as hostname configuration, and even disk formatting.

> Anyways, that's for another day

Absolutely!

arvindshmicrosoft reviewed (commented):
I did a quick "sanity" test with the CentOS image on Azure, just to make sure there are no side effects - there were none.

arvindshmicrosoft commented:
@ctubbsii - how would you like to proceed with this? The only question in my mind is whether (or not) to change the default AMI for EC2. On the Azure side, there are some challenges with broad availability of Fedora / Alma / CentOS Stream images so we will likely retain the older CentOS 7.9 default for Azure for a bit longer. As mentioned above, the changes in this PR do not have any side-effects on CentOS 7.9, so it LGTM.

@keith-turner - would appreciate your input.

ctubbsii commented:
> @ctubbsii - how would you like to proceed with this? The only question in my mind is whether (or not) to change the default AMI for EC2.

My intent was to support Fedora more easily, and to show how similar Fedora and other distros in the same "distro-family" are to the EL7 distros that already work out of the box that we've been using. I was pleasantly surprised to find that EL8 distros were already supported by conditional logic in our ansible playbooks, and aside from the need to use `epel-release` yum repo for that, EL8 distros seems identical to Fedora, which means Fedora hasn't diverged too much from when EL8 was forked from it.

At a minimum, I'd like to merge the conditional logic that I added in the ansible/ directory to better support Fedora. I'd also like to merge the test changes so we're not so tightly coupled to current example config values, but don't lose any test coverage.

I was hoping to defer to the community consensus on whether to change the example config. If the other changes are acceptable, but we want to move that out into a separate PR for further consideration, I can do that. I'll await community consensus on that. I'd be just as happy using an Alma, Rocky, or CentOS 8 Stream AMI by default.

(Note: I also changed the default filesystem format from ext3 to ext4 in this PR, but that was just because I don't see a reason to use the older generation of ext when the newer one is supported. It's not a Fedora-specific change. I didn't test it with CentOS 7, though I'm pretty sure CentOS 7 supports ext4 just fine.)

keith-turner reviewed (approved):
> I was hoping to defer to the community consensus on whether to change the example config. If the other changes are acceptable, but we want to move that out into a separate PR for further consideration, I can do that. I'll await community consensus on that. I'd be just as happy using an Alma, Rocky, or CentOS 8 Stream AMI by default.

If the code likely works w/ Fedora, Rocky, and Alma this PR is fine w/ me.  We can fix problems later as we run into them.
